### PR TITLE
Allow pending changes for `test-checks.sh` for local build

### PR DESCRIPTION
**Pull request by HaoranYi**
#### Problem

During development, it is very nice to run sanity build check (i.e. `test-checks.sh`) with local changes before committing them. However, the `test-check.sh` script requires that there are no uncommitted changes locally. To work around this, developers usually have to commit the pending changes first, then run this script. After finding/fixing errors, commit again and then rerun the scripts ... This is annoying. We should allow the scripts to run with pending changes during development.

#### Summary of Changes

Only check `git diff` for ci build

Fixes #

### CODE REVIEW

File: ci/test-checks.sh [hunk 1]
```diff
@@ -15,6 +44,7 @@ cargoNightly="$(readlink -f "./cargo") nightly"
 
 scripts/increment-cargo-version.sh check
 
+if [ "$local" = false ] ; then
```
Comment (yihau) [inline ci/test-checks.sh:47]: what do you think if we check `CI` environment here? it looks like

```diff
- if [ "$local" = false ] ; then
+ if [ -n "$CI" ] ; then
```
and then we can remove the logic used to parse other arguments.

(btw, using `git diff` to capture uncommitted Cargo.lock changes seems to be a little bit severe)

Comment (HaoranYi) [reply ci/test-checks.sh]: yeah. checking `$CI` is better.  updated.

File: ci/test-checks.sh [hunk 2]
```diff
@@ -7,6 +7,18 @@ set -e
...
+usage: $0 [-h] [-l]
+
+Run sanity check for solana build
+
+  -h              Show help
+  -l              Run sanity check for local repo (allow uncommitted changes)
+
+EOF
+}
+
```
Comment (yihau) [inline ci/test-checks.sh:21]: If we check $CI below, I think we might not need this anymore. 🤔

**Review by yihau — APPROVED** [review]
🫶

File: ci/test-checks.sh (hunk 2 above)
Comment (HaoranYi) [reply ci/test-checks.sh]: yeah. updated.

**Comment by t-nelson** [comment]
can we do this some other way?  i actually want it locally. maybe gate on a `SOLANA_CLI_ALLOW_CARGOLOCK_DESYNC` envvar being defined or something?

**Comment by HaoranYi** [comment]
If you want it locally, can you define $CI locally?

**Comment by t-nelson** [comment]
that will enable a bunch of stuff that doesn't work locally

**Comment by HaoranYi** [comment]
ah. ok. I am fine with introduce the new envvar as long as there is a way to allow pending changes locally. 
Please feel free to make a new PR and commit your change.

**Comment by t-nelson** [comment]
:saluting_face: https://github.com/solana-labs/solana/pull/32798

**Pull request merged**